tests/unit_tests/pyanaconda_tests/modules/payloads/payload/test_module_payload_live_os.py

```diff
@@ -17,103 +17,61 @@
+from pyanaconda.core.util import touch, join_paths
+from pyanaconda.modules.common.errors.payload import IncompatibleSourceError
+from pyanaconda.modules.payloads.constants import SourceType, SourceState
 from pyanaconda.modules.payloads.payload.live_os.live_os import LiveOSModule
 from pyanaconda.modules.payloads.payload.live_os.live_os_interface import LiveOSInterface
+from pyanaconda.modules.payloads.payload.live_image.installation import InstallFromImageTask
+
+from tests.unit_tests.pyanaconda_tests import patch_dbus_publish_object
+from tests.unit_tests.pyanaconda_tests.modules.payloads.payload.module_payload_shared import \
+    PayloadSharedTest
 
 
 class LiveOSInterfaceTestCase(unittest.TestCase):
+    """Test the DBus interface of the Live OS payload."""
 
     def setUp(self):
-        self.live_os_module = LiveOSModule()
-        self.live_os_interface = LiveOSInterface(self.live_os_module)
-
-        self.shared_tests = PayloadSharedTest(payload=self.live_os_module,
-                                              payload_intf=self.live_os_interface)
-
-        self.callback = PropertiesChangedCallback()
-        self.live_os_interface.PropertiesChanged.connect(self.callback)
+        self.module = LiveOSModule()
+        self.interface = LiveOSInterface(self.module)
+        self.shared_tests = PayloadSharedTest(
+            payload=self.module,
+            payload_intf=self.interface
+        )
 
     def _prepare_source(self):
+        """Prepare a default source."""
         return self.shared_tests.prepare_source(SourceType.LIVE_OS_IMAGE)
 
-    def _prepare_and_use_source(self):
-        source = self._prepare_source()
-        self.live_os_module.set_sources([source])
-
-        return source
-
     def test_type(self):
-        self.shared_tests.check_type(PayloadType.LIVE_OS)
+        """Test the type of the payload."""
+        assert self.interface.Type == PAYLOAD_TYPE_LIVE_OS
 
     def test_supported_sources(self):
         """Test LiveOS supported sources API."""
-        assert [SOURCE_TYPE_LIVE_OS_IMAGE] == \
-            self.live_os_interface.SupportedSourceTypes
+        assert self.interface.SupportedSourceTypes == [SOURCE_TYPE_LIVE_OS_IMAGE]
 
     @patch_dbus_publish_object
-    def test_set_source(self, publisher):
+    def test_set_sources(self, publisher):
         """Test if set source API of LiveOS payload."""
         sources = [self._prepare_source()]
-
         self.shared_tests.set_and_check_sources(sources)
 
     @patch_dbus_publish_object
     def test_set_multiple_sources_fail(self, publisher):
         """Test LiveOS payload can't set multiple sources."""
-        paths = [
-            self._prepare_source(),
-            self._prepare_source()
-        ]
-
-        self.shared_tests.set_and_check_sources(paths, exception=IncompatibleSourceError)
-
-    @patch_dbus_publish_object
-    def test_set_when_initialized_source_fail(self, publisher):
```

Comment:
Could you please explain why you are removing these tests?

Reply:
This is already tested in `test_set_when_initialized_source_fail` of the `PayloadBaseInterfaceTestCase` class. The tested functionality is implemented in the `PayloadBase` class and it doesn't change, so it is not necessary to test it with all types of payloads and sources.